lib/matplotlib/scale.py

```diff
@@ -82,6 +82,84 @@ def get_transform(self):
         return IdentityTransform()
 
 
+class ArbitraryTransform(Transform):
+     """
+     A simple transform that takes and arbitrary function for the
+     forward and inverse transform.
+     """
+
+     input_dims = 1
+     output_dims = 1
+     is_separable = True
+     has_inverse = True
+
+     def __init__(self, forward, inverse):
+         """
+         Parameters
+         ----------
+
+         forward: The forward function for the transform
+
+         inverse: The inverse of the forward function.
+         """
+         super().__init__()
+         if callable(forward) and callable(inverse):
+             self._forward = forward
+             self._inverse = inverse
+         else:
+             raise ValueError('arguments to ArbitraryTransform must '
+                              'be functions')
+
+     def transform_non_affine(self, values):
+         return self._forward(values)
+
+     def inverted(self):
+         return ArbitraryTransform(self._inverse, self._forward)
+
+
+class ArbitraryScale(ScaleBase):
+     """
+     Provide an arbitrary scale for the axis.
+     """
+
+     name = 'arbitrary'
+
+     def __init__(self, axis, functions):
+         """
+         Parameters
+         ----------
+
+         axis: the axis for the scale
+
+         functions: (forward, inverse)
+            two-tuple of the forward and inverse functions for the scale.
```

Comment:
See `ArbitraryTransform` for details on the functions.

Reply:
Actually this is more user facing so I changed this to have the more complete info as well.  A bit repetitive, but...